src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -251,57 +262,95 @@ private FormData prepareFormDataSet(FormFields form) {
 
 	/**
 	 * Gets the values for the given {@code field}.
-	 * <p>
-	 * If the field is of submit type it returns its predefined values.
+	 * If the field is of submit type it passes the predefined values to the DefaultValueGenerator and returns its predefined values.
+	 * Gets the default value that the input field, including HTML5 types, should have.
 	 *
 	 * @param field the field
 	 * @return a list with the values
-	 * @see #getDefaultTextValue(FormField)
 	 */
-	private List<String> getValues(FormField field) {
-		if (field.getFormControl().getFormControlType().isSubmit()) {
-			return new ArrayList<>(field.getPredefinedValues());
-		}
+	private List<String> getDefaultTextValue(FormField field) {
 
 		// Get its value(s)
 		List<String> values = field.getValues();
+		String fieldId = field.getName();
+		String defaultValue = "";
+		String finalValue;
+
+		// Create new HashMap 'fieldAttributes' and new list 'definedValues'
+		Map<String, String> fieldAttributes = new HashMap<String, String>();
+		List<String> definedValues = new ArrayList<String>();
+
+		//Store all values in the FormFiled field into the Map 'fieldAttributes'
+		fieldAttributes.putAll(field.getFormControl().getAttributesMap());
+
+		// Places a key, Control Type, for each FormControlType
+		fieldAttributes.put("Control Type", field.getFormControl().getFormControlType().name());
+
+		//If the field has a value attribute present(Predefined value)
+		//Should store the value being submitted to be passed to the ValueGenerator
+		if(field.getFormControl().getAttributesMap().containsKey("value")){
```

Comment:
Could be moved to after the if "Handles Submit Fields" (same reason as the move of `values`).